### DISCUSSION TITLE
Request voor features

### DISCUSSION THREAD

**Discussion by pd5wl** [magicbug/Cloudlog#2345] [Ideas / Feature Requests]
Dear OM,

I only just started using CloudLog a few days a go. After being off the air for some 5 years. I wish I found the software back then. 
I love it. But... (yes i am Dutch, so there is always a but...)

I would like to see and maybe assist in adding 2 features:

1. Award upload (pdf/jpg).

The ability to upload and show awards/certificates and show them on a page. I am a member of FT8DMC and we get our awards as JPG/PDF. I would love to have a place to store them along with all my logs.

FT8DMC has there onw system for getting award, as have many other clubs, so I am not looking to hook on to that. Just a simpel way to store them with my logs and view them (maybe even with a public page.

2. E-mail OQRS reply.
 
I also noticed that a lot of hams send me e-mail with a QSL card. And as i no longer have the Bureau to send then for me (LOTW/QRZ/ClubLog is enough) I would like to ability to reply to these requests via e-mail. (Not going to discus the validity of e-mail QSL card for Awards). I just would like to reply the OM's that send me one or requested one via OQRS. 

Curious to see what you think of this.

With regards,

Wijnand Laros - PD5WL.

**Comment by phl0** [comment]
Hej Wijnand, the former is a completely new feature and indeed sounds nice. 
Re the latter I did some work in the past which was never finished. I plan to have a look at it again and maybe integrate it into the OQRS part by @AndreasK79. However no guarantees as of now :-)

**Reply by pd5wl** [reply]
I am looking in to the simpel part as wel. Award upload.

Currently in the proses of looking trough the code and how you build it. So I can already written parts for the Award uploads.

**Comment by pd5wl** [comment]
For the awards did some thinking.

What do you wanne store:

Memberships.
Unique ID, Club name, Member number, website, Logo of the club (file upload).

Awards.
Unique ID, file location, Award name, Unique ID (from club).

For the front (behind the login in) there could be, Club membership management page, Award Upload page. View membership with a sub showing your awards.. 

Also a public page (enable/disable in the admin/user settings) showing Club memberships + Awards.

When my PC is backup and running (HDD crashed to day.... :(, minor setback) I am gone dive in to you code a bit to see if I can be of any help with this.

73 de PD5WL

**Reply by phl0** [reply]
Sounds good imho. Let us know here if you need any support. I am sure there is a few who can help out.
Looking Forward to your implementation.
